scripts/dts/extract/clocks.py

```diff
@@ -153,6 +153,39 @@ def _extract_consumer(self, node_path, clocks, def_label):
                             clock_label,
                             prop_alias)
 
+                # If the provided clock has a fixed rate, extract its frequency
+                # as a macro generated for the clock consumer.
+                if clock_provider['props']['compatible'] == 'fixed-clock':
+                    clock_prop_name = 'clock-frequency'
+                    clock_prop_label = 'FIXED_CLOCK_FREQUENCY'
+                    if clock_index == 0 and \
```

Comment:
Is it possible for a 'fixed-clock' to be anything but a singleton?

Reply:
=> @anangl

Reply:
I'm not sure if I understood the question correctly but it seems perfectly possible that a device node references several clock nodes from which some are 'fixed-clocks' (for instance, one low- and one high-frequency clock, both with fixed rates). For such cases we need to add indexes to the generated labels.